tdrs-backend/tdpservice/users/api/login.py

```diff
@@ -1,221 +1,145 @@
-        state = openid_authenticity_tracker.get('state', None)
-
-        if 'nonce' not in openid_authenticity_tracker:
-            msg = 'OIDC callback nonce was not found in session `state_nonce_tracker`.'
-            raise SuspiciousOperation(msg)
-
-        nonce = openid_authenticity_tracker.get('nonce', None)
-        validation_keys = {'state': state, 'nonce': nonce}
-        return validation_keys
-
-    """
-    Returns a found users information along with an httpOnly cookie.
-
-    :param self: parameter to permit django python to call a method within its own class
-    :param user: current user associated with this session
-    :param status_message: Helper message to note how the user was found
-    :param id_token: encoded token returned by login.gov/token
-    """
-    def responseInternal(self, user, status_message, id_token):
-        """Respond with an httpOnly cookie to secure the session with the client."""
-        response = Response(
-            {'user_id': user.pk,
-             'email': user.username,
-             'status': status_message
-             }, status=status.HTTP_200_OK)
-        response.set_cookie(
-            'id_token', value=id_token, max_age=None, expires=None, path='/', domain=None, secure=True, httponly=True
-        )
-        return response
+        if not decoded_payload["email_verified"]:  # pragma: no cover
+            return Response(
+                {"error": "Unverified email!"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:  # pragma: no cover
+            # get user from database if they exist. if not, create a new one
+            if "token" not in request.session:
+                request.session["token"] = id_token
+
+            user = CustomAuthentication.authenticate(
+                self, username=decoded_payload["email"]
+            )
+            if user is not None:
+                login(
+                    request,
+                    user,
+                    backend="tdpservice.users.authentication.CustomAuthentication",
+                )
+                datetime_time = datetime.datetime.fromtimestamp(time.time())
+                logger.info(f"Found User:  {user.username} on {datetime_time}(UTC)")
+
+                return response_redirect(user, id_token)
+            else:
+                User = get_user_model()
+                user = User.objects.create_user(decoded_payload["email"])
+                user.set_unusable_password()
+                user.save()
+
+                login(
+                    request,
+                    user,
+                    backend="tdpservice.users.authentication.CustomAuthentication",
+                )
+
+                datetime_time = datetime.datetime.fromtimestamp(time.time())
+                logger.info(f"Created User:  {user.username} at {datetime_time}(UTC)")
+
+                return response_redirect(user, id_token)
+
+        except Exception as e:  # pragma: no cover
```

Comment:
This class method has high cyclomatic complexity and a lot of conditionals. I'd advise refactoring to reduce the length/ depth of this single method and make it more testable.